components/nexusoperations/executors.go

```diff
@@ -74,28 +73,32 @@ func RegisterExecutor(
 	if err := hsm.RegisterExecutor(registry, TaskTypeBackoff.ID, exec.executeBackoffTask); err != nil {
 		return err
 	}
-	return hsm.RegisterExecutor(registry, TaskTypeTimeout.ID, exec.executeTimeoutTask)
+	if err := hsm.RegisterExecutor(registry, TaskTypeTimeout.ID, exec.executeTimeoutTask); err != nil {
+		return err
+	}
+	if err := hsm.RegisterExecutor(registry, TaskTypeCancelation.ID, exec.executeCancelationTask); err != nil {
+		return err
+	}
+	return hsm.RegisterExecutor(registry, TaskTypeCancelationBackoff.ID, exec.executeCancelationBackoffTask)
 }
 
 type activeExecutor struct {
 	ActiveExecutorOptions
 }
 
 func (e activeExecutor) executeInvocationTask(ctx context.Context, env hsm.Environment, ref hsm.Ref, task InvocationTask) error {
-	ns, err := e.NamespaceRegistry.GetNamespaceByID(namespace.ID(ref.WorkflowKey.NamespaceID))
-	if err != nil {
-		return fmt.Errorf("failed to get namespace by ID: %w", err)
-	}
-	ctx, cancel := context.WithTimeout(ctx, e.Config.InvocationTaskTimeout(ns.Name().String()))
-	defer cancel()
-
 	service, err := e.NamespaceRegistry.NexusOutgoingService(namespace.ID(ref.WorkflowKey.NamespaceID), task.Destination)
 	if err != nil {
-		if st, ok := status.FromError(err); ok && st.Code() == codes.NotFound {
+		if errors.As(err, new(*serviceerror.NotFound)) {
```

Comment:
Other places we are using a temp variable for the error pointer; why use `new` here?

Reply:
We've used this pattern in a couple other places. It's fine if you don't need the actual error.